addons/account_edi_ubl_cii/models/account_edi_common.py

```diff
@@ -0,0 +1,592 @@
+            return custom_warning_message or _("The element %s is required on %s.", record, ', '.join(field_names))
+
+        display_field_names = record.fields_get(field_names)
+        if len(field_names) == 1:
+            display_field = f"'{display_field_names[field_names[0]]['string']}'"
+            return _("The field %s is required on %s.", display_field, record.display_name)
+        else:
+            display_fields = ', '.join(f"'{display_field_names[x]['string']}'" for x in display_field_names)
+            return _("At least one of the following fields %s is required on %s.", display_fields, record.display_name)
+
+    def _check_constraints(self, constraints):
+        return [x for x in constraints.values() if x]
+
+    # -------------------------------------------------------------------------
+    # COMMON CONSTRAINTS
+    # -------------------------------------------------------------------------
+
+    def _invoice_constraints_common(self, invoice):
+        constraints = {}
+        # check that there is at least one tax repartition line !
+        for tax in invoice.invoice_line_ids.mapped('tax_ids'):
+            for line_repartition_ids in ['invoice_repartition_line_ids', 'refund_repartition_line_ids']:
+                lines = tax[line_repartition_ids]
+                base_line = lines.filtered(lambda x: x.repartition_type == 'base')
+                if not lines - base_line:
+                    constraints.update({
+                        'tax_repartition_line': _("Taxes should have at least one tax repartition line."),
+                    })
+        # check that there is a tax on each line
+        for line in invoice.invoice_line_ids:
+            if not line.tax_ids:
+                constraints.update({
+                    'tax_on_line': _("Each invoice line should have at least one tax."),
+                })
+        return constraints
+
+    # -------------------------------------------------------------------------
+    # Import invoice
+    # -------------------------------------------------------------------------
+
+    def _import_invoice(self, journal, filename, tree, existing_invoice=None):
+        move_type, qty_factor = self._import_get_document_type(filename, tree)
+        if not move_type or (existing_invoice and existing_invoice.move_type != move_type):
+            return
+
+        invoice = existing_invoice or self.env['account.move']
+        invoice_form = Form(invoice.with_context(
+            account_predictive_bills_disable_prediction=True,
+            default_move_type=move_type,
+            default_journal_id=journal.id,
+        ))
+        invoice_form, logs = self._import_fill_invoice_form(journal, tree, invoice_form, qty_factor)
+        invoice = invoice_form.save()
+        if invoice:
+            if logs:
+                body = "<strong>Format used to import the invoice: " + str(self._description) + "</strong>" \
```

Comment:
translations!

Reply:
strings concatenation like this is not efficient. Use formating instead:
`f"<strong>{_("Format used to import the invoice:")} {self._description}</strong>"`